src/Microsoft.ML.Core/Data/ISchemaBindableMapper.cs

```diff
@@ -30,13 +30,18 @@ public interface ISchemaBindableMapper
     /// This interface is used to map a schema from input columns to output columns. The <see cref="ISchemaBoundMapper"/> should keep track
     /// of the input columns that are needed for the mapping.
     /// </summary>
-    public interface ISchemaBoundMapper : ISchematized
+    public interface ISchemaBoundMapper
     {
         /// <summary>
         /// The <see cref="RoleMappedSchema"/> that was passed to the <see cref="ISchemaBoundMapper"/> in the binding process.
         /// </summary>
         RoleMappedSchema InputRoleMappedSchema { get; }
 
+        /// <summary>
+        /// Gets schema of this mapper's output.
+        /// </summary>
+        Schema Schema { get; }
```

Comment:
>Schema [](start = 15, length = 6)

Maybe OutputSchema?

Reply:
They do not appear to be distinct anywhere in this change, unless I'm very much mistaken. In any case where they were, anyway, you could make it an explicit interface implementation `on ISchemaBoundMapper` since this is not something we want to expose to people).

---
In reply to: [237265683](https://github.com/dotnet/machinelearning/pull/1759#discussion_r237265683) [](ancestors = 237265683,237260730)

Reply:
I am not quite sure if explicit interface implementation is a solution to my problem. At this place ([https://github.com/dotnet/machinelearning/blob/533e18647a77c31f8149ccf2618ccb2e354f8e62/src/Microsoft.ML.Data/Prediction/Calibrator.cs#L538](https://github.com/dotnet/machinelearning/blob/533e18647a77c31f8149ccf2618ccb2e354f8e62/src/Microsoft.ML.Data/Prediction/Calibrator.cs#L538)), the `_predicto`r is an `ISchemaBoundRowMapper`  which contains a `Schema` from `ISchemaBoundMapper` and `OutputSchema` from `OutputSchema`. If we rename this `Schema` to `OutputSchem`a, there is no place where I can make an explicit interface implementation because they conflict inside an Interface. My solution now is to introduce a new interface, called IHasOutputSchema, which contains only one field `Schema OutputSchema { get; }`. Does it make sense to you?

---
In reply to: [237585747](https://github.com/dotnet/machinelearning/pull/1759#discussion_r237585747) [](ancestors = 237585747,237265683,237260730)

Reply:
Now, we do
```csharp
    public interface ISchemaBoundRowMapper : ISchemaBoundMapper, IRowToRowMapper
    {
        /// <summary>
        /// There are two schemas from <see cref="ISchemaBoundMapper"/> and <see cref="IRowToRowMapper"/>.
        /// Since the two parent schema's are identical in all derived classes, we merge them into <see cref="OutputSchema"/>.
        /// </summary>
        new Schema OutputSchema { get; }
    }

```
to unify OutputSchema's from the two parent interfaces.

---
In reply to: [237624643](https://github.com/dotnet/machinelearning/pull/1759#discussion_r237624643) [](ancestors = 237624643,237585747,237265683,237260730)

Reply:
Huzzah!

Reply:
Thanks to your brilliant idea.

---
In reply to: [237949859](https://github.com/dotnet/machinelearning/pull/1759#discussion_r237949859) [](ancestors = 237949859)